components/base-node.js

```diff
@@ -44,6 +46,18 @@ module.exports = {
    },
 
   /**
+   * Standard component error handling - send a message to the error port and throw Error
+   *
+   * @param message message to be sent to error port
+   * @param errorPort handle to the component's error port
+   */
+   handleError: function( message, errorPort ) {
```

Comment:
We may want to (outside this PR) to produce low level components that reflect their technical nature (i.e. they only do one thing) and produce corresponding higher level subgraphs that capture common use cases for them (such as having a logger already attached into their error port or an http-accept attached to an inport).